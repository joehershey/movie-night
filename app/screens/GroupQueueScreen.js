import React from "react";
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableWithoutFeedback,
  SafeAreaView,
} from "react-native";

import { COLORS, STYLES } from "../assets/saved";
import { FontAwesome5 } from "@expo/vector-icons";

import TabBar from "../components/TabBar";
import TopBar from "../components/TopBar";
import Queue from "../components/Queue";

import fetch from "cross-fetch";

function GroupQueueScreen(props) {
  const [isAdmin, setIsAdmin] = React.useState(false);

  React.useEffect(() => {
    const unsubscribe = props.navigation.addListener("focus", () => {
      getUsersAPI();
    });

    return unsubscribe;
  }, [props.navigation]);

  //used to see if the user is an admin
  function getUsersAPI() {
    fetch(props.url + "group/" + props.group_id + "/users", {
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json; charset=utf-8",
        Authorization: "Bearer " + props.token,
      },
      method: "GET",
    })
      .then((response) => response.json())
      .then((responseJson) => {
        members = responseJson;
        for (let i = 0; i < members.length; i++) {
          if (members[i].user_id == props.user_id) {
            setIsAdmin(members[i].is_admin ? true : false); //member isAdmin set true if active user is_admin
            break;
          }
        }
      })
      .catch((error) => {
        console.error(error);
      });
  }

  console.log("on queue");
  return (
    <SafeAreaView style={STYLES.container}>
      {/* Top Bar */}
      <TopBar
        navigation={props.navigation}
        screen="Landing"
        message="Groups"
      ></TopBar>
      {/* Content */}
      <View style={[STYLES.content, { backgroundColor: COLORS.primary }]}>
        <Queue
          navigation={props.navigation}
          url={props.url}
          user_id={props.user_id}
          group_id={props.group_id}
          token={props.token}
          isAdmin={isAdmin}
        ></Queue>
      </View>
      {/* Tabs */}
      <TabBar navigation={props.navigation} screen="queue"></TabBar>
    </SafeAreaView>
  );
}

export default GroupQueueScreen;
