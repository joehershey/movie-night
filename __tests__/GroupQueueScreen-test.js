import GroupQueueScreen from '../app/screens/GroupQueueScreen';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Test Group Queue Screen', () => {
    beforeEach( async () => {
        global.tree = renderer.create(<GroupQueueScreen />).toJSON();
    });

    it("Group queue screen renders properly", async () => {
        await expect(global.tree).toMatchSnapshot();
    });
});