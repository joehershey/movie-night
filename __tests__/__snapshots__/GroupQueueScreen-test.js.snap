// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Group Queue Screen Group queue screen renders properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1d1f1e",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1d1f1e",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "margin": 5,
          }
        }
      >
        Groups
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "paddingBottom": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginBottom": 0,
            "width": 100,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 11,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "margin": 20,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "fontSize": 25,
              "margin": 10,
            }
          }
        >
          Under Construction
        </Text>
        <Text />
      </View>
      <Text
        style={
          Object {
            "fontSize": 25,
            "fontStyle": "italic",
            "margin": 30,
            "textAlign": "center",
          }
        }
      >
        Coming in Iteration 2!
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#1d1f1e",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "normal",
            "margin": 5,
          }
        }
      >
        Schedule
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#166fdb",
            "fontWeight": "bold",
            "margin": 5,
          }
        }
      >
        Queue
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "normal",
            "margin": 5,
          }
        }
      >
        Search
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "normal",
            "margin": 5,
          }
        }
      >
        Settings
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
